go/mysql/constants.go

```diff
@@ -542,14 +543,7 @@ func IsNum(typ uint8) bool {
 func IsConnErr(err error) bool {
 	if sqlErr, ok := err.(*SQLError); ok {
 		num := sqlErr.Number()
-		// ServerLost means that the query has already been
-		// received by MySQL and may have already been executed.
-		// Since we don't know if the query is idempotent, we don't
-		// count this error as connection error which could be retried.
-		if num == CRServerLost {
```

Comment:
This case is gone now, so a CRServerLost will return IsConnErr = true, before it was false. Was this an intentional change?

Reply:
Yes. The change now uses the ctx.Done() to detect timeout errors, which is more correct. This allows us to treat all broken connections the same way.

Reply:
Also, we use `resultSent` to decide if a streaming query can be retry, which is also more correct.